src/mixins/labels.js

```diff
@@ -95,6 +96,71 @@ export const LabelsMixin = (superclass) =>
             });
         }
 
+        /**
+         * This function makes a hash map from atom names.
+         * Used it for sorting all atoms by names and then quckly get it for creating a label points
+         * @returns {Object.<string, Array.<Three.InstancedMesh>>} atomsHashMap
+         */
+        makeHashMapFromAtomNames() {
+            const atomsHashMap = {};
+            this.structureGroup.children.forEach((group) => {
+                if (group.name !== ATOM_GROUP_NAME) return;
+
+                group.children.forEach((atom) => {
+                    if (atom instanceof THREE.Mesh) {
+                        const text = atom.name.split("-")[0];
+                        if (!atomsHashMap[text]) {
+                            atomsHashMap[text] = [atom];
+                            return;
+                        }
+                        atomsHashMap[text].push(atom);
+                    }
+                });
+            });
+
+            return atomsHashMap;
+        }
+
+        /*
+         * function that creates label sprites as points.
+         * If we want to use a lot of labels and don't want to have a huge impact
+         * from rendering scene we should use Three.Points or Three.InstancedMesh.
+         * https://threejs.org/docs/#api/en/objects/Points
+         * https://threejs.org/docs/?q=instanced#api/en/objects/InstancedMesh
+         */
+        createLabelSpritesAsPoints() {
+            if (this.#labels.length) {
+                this.#labels.forEach((label) => this.structureGroup.remove(label));
+            }
+
+            const atomsHashMap = this.makeHashMapFromAtomNames();
+            Object.entries(atomsHashMap).forEach(([key, atoms]) => {
+                const texture = this.getLabelTextTexture(key);
+                const vertices = [];
+                atoms.forEach((atom) => {
+                    const position = new THREE.Vector3().setFromMatrixPosition(atom.matrixWorld);
+                    const { x, y, z } = position;
+                    vertices.push(x, y, z);
+                });
+                const geometry = new THREE.BufferGeometry();
+                geometry.setAttribute("position", new THREE.Float32BufferAttribute(vertices, 3));
+                const material = new THREE.PointsMaterial({
+                    size: 1.5,
+                    map: texture,
+                    depthTest: true,
+                    depthFunc: THREE.NotEqualDepth,
+                    transparent: true,
```

Comment:
Should we move the above config to settings?

Reply:
Do you mean PointsMaterial params?